packages/firebase/index.d.ts

```diff
@@ -7030,17 +7089,81 @@ declare namespace firebase.messaging {
     setBackgroundMessageHandler(
       callback: (payload: any) => Promise<any> | void
     ): void;
+
     /**
      * To use your own service worker for receiving push messages, you
      * can pass in your service worker registration in this method.
      *
      * @param registration The service worker
      *   registration you wish to use for push messaging.
+     * 
+     *     @deprecated Use getToken(options?: {
+      vapidKey?: string;
+      serviceWorkerRegistration?: ServiceWorkerRegistration;
+    }): Promise<string>;.
      */
+
     useServiceWorker(registration: ServiceWorkerRegistration): void;
+
+    /**
+     * @deprecated Use getToken(options?: {
+      vapidKey?: string;
+      serviceWorkerRegistration?: ServiceWorkerRegistration;
+    }): Promise<string>;.
+     */
     usePublicVapidKey(b64PublicKey: string): void;
   }
 
+  /**
+   * Message payload that contains the notification payload that is represented with {@link firebase.Messaging.NotificationPayload} and the data payload that contains an arbitrary number of key-value pairs sent by developers through the {@link https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#notification Send API}
+   */
+  export interface MessagePayload {
+    /**
+     * See {@link firebase.Messaging.NotificationPayload}.
+     */
+    notification?: NotificationPayload;
+
+    /**
+     * Arbitrary key/value pairs.
+     */
+
+    data?: { [key: string]: string };
+
+    /**
+     * See {@link firebase.Messaging.FcmOptions}.
+     */
+
+    fcmOptions?: FcmOptions;
+  }
+
+  /**
+   * Options for features provided by the FCM SDK for Web. See more {@link https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#webpushfcmoptions}.
```

Comment:
Could we add a bit more context here as to why these might be necessary. A `for example` may be useful.

Reply:
Got it. I added to the comment for the specific parameter